lib/google_maps_geocoder/google_maps_geocoder.rb

```diff
@@ -104,19 +129,16 @@ def json_from_url(url)
     ActiveSupport::JSON.decode response.body
   end
 
-  def handle_exception(google_response_json)
-    status = google_response_json['status']
-    # no exception
-    return if status == 'OK'
-
-    message = GoogleMapsGeocoderErrorHandler::GeneralGoecodingError\
-              .new(google_response_json).message
+  def handle_error
+    status = @json['status']
+    message = GeocodingError.new(@json).message
 
     # for status codes see https://developers.google.com/maps/documentation/geocoding/intro#StatusCodes
-    raise GoogleMapsGeocoderErrorHandler::ZeroResultsError,\
```

Comment:
instead of repeating this 5 times I added some "magic" with active_support inflections help